File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def depth(self,root):
        if(root):
            return 1 + max(self.depth(root.left),self.depth(root.right))
        return 0
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if(root):
            current = (abs(self.depth(root.right)-self.depth(root.left))<=1)
            return current and self.isBalanced(root.left) and self.isBalanced(root.right)
        return True

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root){
            int rdepth = 1 + maxDepth(root->right);
            int ldepth = 1 + maxDepth(root->left);
            return max(ldepth,rdepth);
        }
        else{
            return 0;
        }
    }
};
